src/kp2bw/bw_serve.py

```
@@ -0,0 +1,574 @@
+        return session or None
+
+    def _start_serve(self, session: str | None = None) -> None:
+        """Spawn ``bw serve --port <port> --hostname localhost``."""
+        cmd = ["bw", "serve", "--port", str(self._port), "--hostname", "127.0.0.1"]
+        env: dict[str, str] | None = None
+        if session:
+            env = {**os.environ, "BW_SESSION": session}
+        logger.log(
+            VERBOSE,
+            f"Starting bw serve on port {self._port} "
+            f"(BW_SESSION={'set' if session else 'not set'})",
```

Ruff (0.15.2)

[warning] Logging statement uses f-string

(G004)